Sources/Subs-Package.php

```diff
@@ -246,6 +246,11 @@ function read_zip_file($file, $destination, $single_file = false, $overwrite = f
 		// This may not always be defined...
 		$return = array();
 
+		// Some hosted unix platforms require an extension; win may have .tmp & that works ok
+		if (strtolower(substr($file, -4)) != '.zip' && strtolower(substr($file, -4)) != '.tmp')
+			if (@rename($file, $file . '.zip'))
+				$file = $file . '.zip';
+
```

Comment:
This would be more reliable:
```
if (!in_array(strtolower(pathinfo($file, PATHINFO_EXTENSION)), array('zip', 'tmp')) && @rename($file, $file . '.zip'))
	$file = $file . '.zip';
```

Reply:
Actually, I would think that it's probably better to skip the .tmp option entirely and just force a rename to a zip extension in all cases.
```
if (strtolower(pathinfo($file, PATHINFO_EXTENSION) !== 'zip' && @rename($file, $file . '.zip'))
	$file = $file . '.zip';
```